gradio/cli/commands/reload.py

```diff
@@ -96,10 +100,23 @@ def _setup_config(
             message += f" '{wd}'"
 
     print(message + "\n")
+    reload_sources: dict[str, str] = {}
+
+    # copy the watched directories to temp directories
+    # so that we can reload the modules after parsing out
+    # the gr.no_reload contexts
+    for s in watching_dirs:
+        temp_dir = tempfile.mkdtemp()
+        for file in Path(s).rglob("*.py"):
+            src = file
+            dst = Path(temp_dir) / file.relative_to(s)
+            dst.parent.mkdir(parents=True, exist_ok=True)
+            shutil.copy(src, dst)
```

Comment:
This will lead to interesting performance characteristics if the the tempdir ends up on a different file system, I'm sure.

Reply:
I realized we don't actually need to do the copying